src/Physics/Multinucleon/XSection/MECScaleVsW.cxx

```diff
@@ -0,0 +1,159 @@
+  double W_min = sqrt( pow(Mn,2) + 2*Mn*fW1_Q0Q3_limits.Eval(Q3) - pow(Q3,2) + pow(fW1_Q0Q3_limits.Eval(Q3),2) ) ;
+  double W_max = sqrt( pow(Mn,2) + 2*Mn*Q0 ) ; // Imposing Q2 = 0 
+  
+  // Insert phase space limits:
+  weight_type_map w_map = fWeightsMap ; 
+  w_map.insert( weight_type_pair( W_max, fDefaultWeight ) ) ;
+  w_map.insert( weight_type_pair( W_min, fDefaultWeight ) ) ;
+
+  return w_map ; 
+} 
+//_________________________________________________________________________
+    
+double MECScaleVsW::ScaleFunction( double W, weight_type_pair min, weight_type_pair max ) const 
+{
+  // This function is responsible to calculate the scale at a given W
+  // It interpolates the value between scale_min (W_min) and scale_max (W_max) linearly 
+  // Get Wpoints from pair:
+  double W_min = min.first ; 
+  double W_max = max.first ; 
+  // Get scale coeficients at W_min and W_max:
+  double scale_min = min.second ; 
+  double scale_max = max.second ; 
+  
+  double scale = ( scale_min - scale_max ) * ( W - W_min ) / ( W_min - W_max ) + scale_min ; 
+
+  return scale ; 
+
+} 
+
+//_________________________________________________________________________
+
+void MECScaleVsW::LoadConfig(void)
+{
+  bool good_config = true ; 
+  // Reset members
+  fWeightsMap.clear(); 
+  fW1_Q0Q3_limits = nullptr ; 
+
+  std::vector<double> Weights, WValues ;
+  GetParam( "MECScaleVsW-Default-Weight", fDefaultWeight ) ;
+  GetParamVect( "MECScaleVsW-Weights", Weights ) ; 
+  GetParamVect( "MECScaleVsW-WValues", WValues ) ; 
+
+  if( Weights.size() != WValues.size() ) {
+    good_config = false ; 
+    LOG("MECScaleVsW", pERROR) << "Entries don't match" ;
+    LOG("MECScaleVsW", pERROR) << "Weights size: " << Weights.size() ;
+    LOG("MECScaleVsW", pERROR) << "WValues size: " << WValues.size() ;
+  }
+
+  // Store weights and WValues in map:
+  for( unsigned int i = 0 ; i<Weights.size() ; ++i ) {
+    fWeightsMap.insert( weight_type_pair( WValues[i], Weights[i] ) ) ;
+  }
+
+  std::vector<double> limit_Q0, limit_Q3 ;
+  if( GetConfig().Exists("MECScaleVsW-LowerLimitQ0") ) {
+    GetParamVect( "MECScaleVsW-LowerLimitQ0", limit_Q0 ) ;
+  } else {
+    good_config = false ; 
+    LOG("MECScaleVsW", pERROR) << "Lower limit of the phase space for Q0 is not specified" ;
+  }
+
+  if( GetConfig().Exists("MECScaleVsW-LowerLimitQ3") ) {
+    GetParamVect( "MECScaleVsW-LowerLimitQ0", limit_Q3 ) ;
+  } else {
+    good_config = false ; 
+    LOG("MECScaleVsW", pERROR) << "Lower limit of the phase space for Q3 is not specified" ;
+  }
+
+  if( limit_Q0.size() != limit_Q3.size() ) {
+    good_config = false ; 
+    LOG("MECScaleVsW", pERROR) << "Entries don't match" ;
+    LOG("MECScaleVsW", pERROR) << "Lower limit for Q0 size: " << limit_Q0.size() ;
+    LOG("MECScaleVsW", pERROR) << "Lower limit for Q3 size: " << limit_Q3.size() ;
+  }
+
+  fW1_Q0Q3_limits = TSpline3("fW1_Q0Q3_limits",&limit_Q0[0],&limit_Q3[0],limit_Q0.size());
```

Comment:
We are now in the 3rd millennium and we are allowed to use proper code. Don't hack into the pointers, use 
`std::vector::data()` instead of `& vector[0]`